src/module.c

```diff
@@ -85,6 +85,17 @@ jl_binding_t *jl_get_binding_wr(jl_module_t *m, jl_sym_t *var)
     return *bp;
 }
 
+// return module of binding
+DLLEXPORT jl_module_t *jl_get_module_of_binding(jl_module_t *m, jl_sym_t *var)
+{
+    jl_binding_t **bp = (jl_binding_t**)ptrhash_bp(&m->bindings, var);
+    if (*bp == HT_NOTFOUND) {
+        jl_errorf("Error: symbol :%s is not bound.", var->name); 
+    }
+    jl_binding_t *b = jl_get_binding((*bp)->owner, var);
+    return b->owner;
+}
+
```

Comment:
`jl_get_binding` might return NULL if the symbol is unbound. You should only need to call `jl_get_binding`, check for NULL, and otherwise return `b->owner`. The `ptrhash_bp` call is unnecessary and there are other conditions you would need to check (duplicating `jl_get_binding` -- see its definition in `modules.c`)